Client/Frontend/Browser/Tab.swift

```diff
@@ -251,11 +255,38 @@ class Tab: NSObject {
     }
 
     deinit {
+        debugTabCount -= 1
+
+        #if DEBUG
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+        func checkTabCount(failures: Int) {
+            // Need delay for pool to drain.
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+                if appDelegate.tabManager.tabs.count == debugTabCount {
+                    return
+                }
+
+                if failures > 2 { // If the check fails repeatedly, log an error message.
+                    Sentry.shared.send(message: "Tab init/deinit imbalance, possible memory leak.")
+                } else {
+                    checkTabCount(failures: failures + 1)
+                }
+            }
+        }
+        checkTabCount(failures: 0)
+        #endif
+    }
+
+    func close() {
+        webView?.removeObserver(self, forKeyPath: KVOConstants.URL.rawValue)
+
         if let webView = webView {
-            webView.removeObserver(self, forKeyPath: KVOConstants.URL.rawValue)
             tabDelegate?.tab?(self, willDeleteWebView: webView)
         }
+
         contentScriptManager.helpers.removeAll()
+        webView?.navigationDelegate = nil
```

Comment:
Is there any reason to add `webView?.removeFromSuperview()` here? Obviously it would be a fallback in case it wasn't removed. I couldn't think of a reason to add it.

Reply:
ok this code is already somewhat *defensive* in nature, so an added line of defence makes sense here